### ISSUE TITLE
Need a tutorial for my own image

### ISSUE THREAD

**Issue by Pascal-bucketbreaker** [voxelmorph/voxelmorph#581]
**Task** (what are you trying to do/register?)

I am using it to register my own image, but it can only output image for 256*256.

**What have you tried**

First, I use the pretrained model from colab. It works correctly, but can only output 256*256 image. 
However, my image is 1000*1000, and the resolution is vital to my task, so I try to train the model on my own, but I get out-of-memory.(With Quadro P5000, 16GB RAM)
So how can I use the method to register my own image?

**Details of experiments**

I tried to train 256*256 model on my own ,but it makes nonsense: the loss keeps 0.0303, and the result seems exactly the same to the input.

**Comment by adalca** [comment]
You could try using a model with fewer channels at the highest resolution (e.g. 8 instead of 16) as that probably takes the most memory. You should also pad your image to 1024 to avoid some silly sizing constraints. Finally, make sure your data is normalized in [0, 1].

**Comment by Pascal-bucketbreaker** [comment]
![image](https://github.com/voxelmorph/voxelmorph/assets/46492746/610c76b0-cfd9-4a9b-8e60-71b411ff4dba)
![image](https://github.com/voxelmorph/voxelmorph/assets/46492746/0cc00240-4239-4e3c-8499-3595080d67d9)
![image](https://github.com/voxelmorph/voxelmorph/assets/46492746/5c1b8eb8-30c1-41a7-871b-9c749eb61d86)
However,the input should be [256,256], output should be[256,256,2]
Why does is not match?

**Comment by adalca** [comment]
I'm sorry, can you expand on your question? I don't understand what you are asking.

your code above creates a very large model! `([256] * 4, [256] * 8 )` is much larger than most models we use.

**Comment by Pascal-bucketbreaker** [comment]
But the code above is from your tutorial motebook ,https://colab.research.google.com/drive/1WiqyF7dCdnNBIANEY80Pxw_mVz4fyV-S?usp=sharing#scrollTo=DhZ92bohPXNf
I just cannot understand what the model parameter indicates

**Comment by Pascal-bucketbreaker** [comment]
How can I reduce the channel  number? And how does` ([256] * 4, [256] * 8 )`mean? If I want to process image of `1024*1024`, what I should do?

**Comment by adalca** [comment]
the code above, which uses 32 features (channels) per convolutions, may fit in your memory. 
```
 [32, 32, 32, 32],         # encoder features
 [32, 32, 32, 32, 32, 16]  # decoder features
```
It seems like you changed this to 256 features per channel (this is what `nb_unet_features` refers to), which is a very large model.

the number of features/channels won't be related to your image size.  To work with images of size 1024x1024, I would keep the encoder/decoder to 32 (or 16 for memory efficiency), and what you want to change is the `inshape` variable -- this is the one that controls the size of the image (and original input features, so maybe [1024, 1024, 2])
